src/decimal.rs

```diff
@@ -1514,6 +1514,122 @@ impl Decimal {
             flags: flags(self.is_sign_negative(), quotient_scale),
         })
     }
+
+    /// The estimated exponential function, e<sup>x</sup>, rounded to 8 decimal places. Stops
+    /// calculating when it is within tolerance is roughly 0.000002 in order to prevent
+    /// multiplication overflow.
+    fn exp(&self) -> Decimal {
+        if self == &Decimal::zero() {
+            return Decimal::one();
+        }
+
+        let tolerance = Decimal::new(2, 7);
```

Comment:
Do we ever see this being something that is passed in as an argument? i.e. would it make sense to have two functions for this?

Reply:
I could see it happening, I'll go ahead and create a version with `tolerance` as a argument. Its pretty tricky to hit a nice balance without causing multiplication to overflow though.

Reply:
OK I added a version `exp_with_tolerance` and tests for it.